### DISCUSSION TITLE
Why doesn't redux-toolkit export the thunk middleware

### DISCUSSION THREAD

**Discussion by azangru** [reduxjs/redux-toolkit#1147] [General]
So redux-toolkit [imports and re-exports](https://github.com/reduxjs/redux-toolkit/blob/master/src/index.ts#L18) ThunkAction and ThunkDispatch types from redux-thunk, and it also adds redux-thunk as a default middleware when configuring redux store. Why doesn't it re-export the thunk middleware itself? It might help ever so slightly when setting up a mock redux store for testing purposes: as of now the user would also need to install redux-thunk separately to set up the mock store as [described](https://redux.js.org/recipes/writing-tests) in the redux docs:

```
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk' // <-- why not be able to import thunk from redux-toolkit directly?

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)
```

**Comment by phryneas** [comment]
Honestly, it's just not really a use case we had in mind as we just use the "real" store for testing (at least I know I am).
There just isn't a lot of benefit from using `redux-mock-store` any more.

If you need the actions that were dispatched, just add an additional reducer in your test environment like 
```ts
function actionsReducer(state = [], action) { return [...state, action] }
``` 
and you are rid of an additional dependency.

Also, most React tests nowadays are more integration-like, so many people don't even test reducers in isolations at all.

We'll probably add a few more testing-related things in a future update of RTK. Ideas are very welcome :)
